client/src/lib/api/UserAPI.ts

```diff
@@ -1,12 +1,19 @@
 import config from '@/lib/config';
-import type { UserPatches, PatchUserRequest } from '@/lib/types/apiRequests';
+import type { UserPatches, PatchUserRequest, LoginRequest } from '@/lib/types/apiRequests';
 import type {
   PrivateProfile,
   GetCurrentUserResponse,
   PatchUserResponse,
+  LoginResponse,
 } from '@/lib/types/apiResponses';
 import axios from 'axios';
 
+export const login = async (email: string, password: string): Promise<string> => {
+  const requestBody: LoginRequest = { email, password };
+  const response = await axios.post<LoginResponse>('/api/login', requestBody);
+  return response.data.token;
+};
```

Comment:
I think Next.js has support for server actions, where you can add `'use server'` to the beginning of a function and NextJS will turn it into the same kind of indirection you're doing here. I haven't used it before though, and this works too